### DISCUSSION TITLE
Broken link: scripts/create_venv.sh is not found

### DISCUSSION THREAD

**Discussion by aalexandersson** [moj-analytical-services/splink#1295] [Q&A] [answered]
How to update an existing installation of splink?

I am a beginner Python user. I have used splink 3.4.3 and a related requirements.txt file (e.g., with `duckdb==0.6.0`) in a virtual environment for Python 3.11 on Windows for 6+ months. How do I update to the new [v3.9.0](https://github.com/moj-analytical-services/splink/releases/tag/v3.9.0)? I want to update because the new version has a new critical feature to me, Damerau-Levenshtein comparator in `duckdb==0.8.0`.

The current instructions at [**https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/building_env_locally.html**](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/building_env_locally.html) state

> To automatically create a simple virtual environment using venv, simply run [this script](https://github.com/moj-analytical-services/splink/scripts/create_venv.sh):
> `source scripts/create_venv.sh`

I tried running the command in Windows PowerShell but it gives me this error message:
```
The term 'scripts/create_venv.sh' is not recognized as the name of a cmdlet, function, script file, or operable
program.
```

Also, the web page has a **broken link**. It only has the content `Not Found`. Maybe this is the correct URL with the script?:
[https://github.com/moj-analytical-services/splink/blob/master/scripts/create_venv.sh](https://github.com/moj-analytical-services/splink/blob/master/scripts/create_venv.sh)

If that is the script, how do I run `scripts/create_venv.sh` from Windows PowerShell to update splink? 

I am trying to learn about the script. In the meanwhile, my "old" method of editing requirements.txt for updating to splink==3.9.0 and duckdb==0.8.0 still worked:

`pip3 install -r requirements.txt`

Update: The link is broken. Otherwise my question seems to be a duplicate of [issue 1031](https://github.com/moj-analytical-services/splink/issues/1031). I updated the discussion to reflect this.

**Comment by RossKen** [comment, marked as answer]
Hey @aalexandersson, your "old method" is the correct way of doing it. The `source scripts/create_venv.sh` command is intended for people developing splink, not for users - so for use when you clone the splink repo to add a feature/bug fix etc.   Since the core team work on macs, it's intended to be run in unix-based environment, hence the errors you're seeing.

You are correct that Damerau-Levenshtein requires duckdb 0.8.0, however we haven't made that a strict requirement for installing splink. This is because we know of users who have struggled to install duckdb in their specific environments and given Damerau-Levenshtein is an optional feature we decided not to force everyone to upgrade to 0.8.0. Anyone requiring the most up to date version can then upgrade their duckdb install as they wish to access the latest features. However, I admit that we could have documented this better.

**Comment by aflaxman** [comment]
In case it is useful to others, I just ran into this on Google Colab yesterday, and was able to use D-L by starting my notebook with the following cell
```
!pip install --upgrade pseudopeople splink duckdb
```
